src/main/java/com/greenfox/poker/service/TokenService.java

```diff
@@ -0,0 +1,69 @@
+package com.greenfox.poker.service;
+
+import com.greenfox.poker.model.PokerUser;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MissingClaimException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.impl.crypto.MacProvider;
+
+import io.jsonwebtoken.lang.Assert;
+import java.security.Key;
+import java.util.*;
+import java.lang.*;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TokenService {
+
+  private Key key;
+
+  public String generateToken(PokerUser pokerUser) {
+    HashMap<String, Object> claims = new HashMap<>();
+    //   claims.put("id", pokerUser.getId());
+    claims.put("username", pokerUser.getUsername());
+    return generateToken(claims);
+  }
+
+  private String generateToken(HashMap<String, Object> claims) {
+    this.key = MacProvider.generateKey();
+    String token = Jwts.builder()
+        .setClaims(claims)
+        .signWith(SignatureAlgorithm.HS256, key)
+        .compact();
+    return token;
+  }
+
+  private Claims getClaimFromToken(String token) {
+    Claims claims;
+    claims = Jwts.parser()
+        .setSigningKey(key)
+        .parseClaimsJws(token)
+        .getBody();
+    return claims;
+  }
+
+  public String getUsernameFromToken(String token) {
```

Comment:
Instead of these separate function, you should create a function that return with a User object.
- get the id from the token
- fetch the user by its id from the DB
- check if the username is matching in the token and in the DB object
- return with the user object